README
------------------------------------------------------------------------------
ATLAS LaTeX class, style files and templates to typeset notes and papers.
See ChangeLog or SVN log for history of changes.

Responsible: Ian Brock (Ian.Brock@cern.ch)
------------------------------------------------------------------------------

The following template main files exist:
  atlas-document.tex		Generic ATLAS document
  atlas-document-texmf.tex	Generic ATLAS document for a central atlaslatex installation
  atlas-draft-cover.tex		Make a standalone cover for an ATLAS draft
  atlas-preprint-cover.tex	Make a standalone cover for an ATLAS CERN preprint
  atlas-auxmat.tex			A front page for auxiliary material

The ATLAS document class (atlasdoc.cls) and style files can be found in 
the latex directory. The following main style files exist:
  atlasbiblatex.sty		Reference style adjustments for biblatex
  atlascover.sty		Make a cover (CONF note, CERN preprint, ATLAS draft)
  atlascontribute.sty	List of contributors (and authors) for a document
  atlaspackage.sty		Standard packages used in ATLAS documents
  atlasphysics.sty		Useful definitions. This file simply inputs others.
						Options can be used to specify which should be included.

Documentation can be found in the doc directory.
  atlas-paper.pdf		Guide to the content and style of ATLAS papers
  atlas-latex.pdf		Guide to the use of the ATLAS document templates and styles
  atlas-bibtex.pdf		Guide to references and BibTeX in ATLAS
  atlas-physics.pdf		Symbols defined in atlasphysics.sty
  atlas-tables.pdf		Some guidelines and help for tables

More detailed information about the package can be found under:
https://twiki.cern.ch/twiki/bin/view/AtlasProtected/PubComLaTeX

The general idea is that you create a separate directory tree for each document that
you want to produce which contains the contents of this package.
It is assumed that all style files are in a directory latex, which is a subdirectory of 
the one in which the main document sits.
The latex subdirectory can of course be a link to a central style directory.
You can use the \ATLASLATEXPATH variable to specify an arbitrary directory.

To make a new document give the command:
  make new [BASENAME=mydocument] [TEXLIVE=YYYY]
The TeX Live version is set to 2013 by default.
This version number should be fine for newer versions of TeX Live 
and also for an up-to-date MikTeX 2.9 installation.
  make help
gives you a bit more assistance on which make targets exist.

The most common FAQ I get is why atlaslatex does not just compile "out of the box"?
If you are running on lxplus for it to work, you MUST set your PATH correctly:
  export PATH=/afs/cern.ch/sw/XML/texlive/latest/bin/x86_64-linux:$PATH

A users guide to the templates can be found in doc/atlas_latex.pdf. You can produce
this document yourself (and thus test that your LaTeX setup is working)
by giving the commands:
  cd doc/atlas_latex
  make
or
  cd doc/atlas_latex
  pdflatex atlas_latex
  biber    atlas_latex
  pdflatex atlas_latex
  pdflatex atlas_latex

Three other make targets are:
  make clean				Cleans up intermediate files
  make cleanpdf				Remove output pdf file
  make cleanall				Also cleans up output pdf file


ChangeLog
2021-03-26  Ian Brock <Ian.Brock@cern.ch>
	* Tag as version 11.0.0
	* Use of ChangeLog is discontinued. This file will be deleted in version 11.1.0.

2021-02-09  Ian Brock <Ian.Brock@cern.ch>
	* Tag as version 10.3.0
	* Updated conference entries to follow fixed scheme with use of eventdate and venue fields.
	* Documented this in atlas_bibtex.pdf.
2021-02-08  Ian Brock <Ian.Brock@cern.ch>
	* Tag as version 10.2.0
	* December update of publications.
	* longtable is now an option for atlaspackage.sty and atlascontribute.sty.
	* All history now included in CHANGELOG.md.
2021-01-08  Ian Brock  <Ian.Brock@cern.ch>
	* Tag as version 10.1.0
	* November update of publications.
	* Fixed atlaslatex-update.py self-update to only have to be run once.
	* tocdepth fixed for TeX Live 2013 (option does not work).
	* Get rid of latex link in journal directory.
2020-11-23  Ian Brock  <Ian.Brock@cern.ch>
	* Tag as version 10.0.0
	* New style file atlaslatexpath.sty to set search path for style files and document class.
	* New script atlaslatex_2020.sh to make adjustments to main files for TeX Live 2020.
	* Changed version numbering from 09-03-00 to 10.0.0 (follows Semantic Versioning).
	* Templates and documentation switched to use of atlaslatexpath.sty.
	* atlaslatex_update.sh script can self update and checks for use of \ATLASLATEXPATH macro.
	* The macro \ATLASLATEXPATH should no longer be necessary or be used.
	* Removed texmf targets in Makefile, as regular targets work in the same way.
	* Block option was being ignored in atlasbiblatex.sty as of version 07-01-00.
	* Remove a duplicate reference from ATAS-SUSY.bib.

2020-10-15  Ian Brock  <Ian.Brock@cern.ch>
	* Tag as version 09-02-00.
	* August update of publications.
	* Add JINST as a journal template.
2020-09-09  Ian Brock  <Ian.Brock@cern.ch>
	* Tag as version 09-01-00.
	* June and July update of publications.
	* Complete set of TDRs now included. Changed citation ATLAS-TDR-2010-19 to ATLAS-TDR-19.
	* A number of small updates to the default detector description from David Stoker.
	* Add usage of \tablenum for tables.
	* Use only \AtlasTitle macro and not \title macro when using elsarticle class.
2020-07-23  Ian Brock  <Ian.Brock@cern.ch>
	* Tag as version 09-00-03.
	* Only TeX Live 2013 and later now supported.
	* Major additions to ATLAS-useful.bib from PMG group.
	* Several generators added to atlasmisc.sty.
	* Add atlastodo.sty for to-do notes.
	* Add to-do notes package to templates.
	* Add option svgnames to xcolor.
	* Move atlasdoc2.cls and atlascover2.sty to obsolete directory.

2020-06-26  Ian Brock  <Ian.Brock@cern.ch>
	* Tag as version 08-05-00.
	* April and May 2020 update of publications.
2020-05-17  Ian Brock  <Ian.Brock@cern.ch>
	* Tag as version 08-04-00.
	* March 2020 update of publications including CMS backlog.
	* Updated acknowledgements citation.
	* Add fix for line numbering around AMS Math LaTeX environments.
	* Number lines inside equations by default (mathlines option).
2020-04-06  Ian Brock  <Ian.Brock@cern.ch>
	* Tag as version 08-03-00.
	* December 2019 update of publications.
	* Add bookmark package by default.
2020-03-15  Ian Brock  <Ian.Brock@cern.ch>
	* Tag as version 08-02-00.
	* October and November updates of publications.
	* Add cleveref package as a default package to atlaspackage.
	* Add physics package when passing option full to atlaspackage.
	* Add unicode and psdextra options for hyperref.
	* New macro \AtlasOrcid for ORCID.
	* ORCID can be included as an optional argument in \AtlasAuthorContributor.
	* Links in authors will be black. This is used for ORCID links.
2019-11-26  Ian Brock  <Ian.Brock@cern.ch>
	* Tag as version 08-01-00.
	* Updated acknowledgements.
	* Add atlaslogo and cernlogo options to turn on/off showing the ATLAS and CERN logos.
	* Change default percentspace option in atlaspackage.sty to false.
2019-11-17  Ian Brock  <Ian.Brock@cern.ch>
	* Tag as version 08-00-00.
	* August and September updates of publications + a few fixes.
	* Rename macro \Ref to \Refn and \Refs to \Refns, as hyperref defines \Ref in TeX Live 2019.
	* Add a first CI for atlaslatex.
	* Updated acknowledgements.

2019-09-13  Ian Brock  <Ian.Brock@cern.ch>
	* Tag as version 07-05-01.
	* Remove usage of tocloft package, as it can conflict with KOMA-Script.
	* Set default table of contents depth to section and add tocdepth option.
	* July update of publications (except CMS).
	* Fix up a number of references in ATLAS-useful.bib and add a few new ones.
	* Add some, but not all, of the ATLAS TDRs to ATLAS.bib.
	* Small updates to SUSY templates.
2019-05-21  Ian Brock  <Ian.Brock@cern.ch>
	* Tag as version 07-04-00.
	* Tweak the logic of the backref option.
	Now off by default, except for draft documents.
2019-05-09  Ian Brock  <Ian.Brock@cern.ch>
	* Tag as version 07-03-00.
	* April update of publications.
	* Tweak atlaslatex_update.sh to not overwrite help text with BASENAME.
	* A few fixes to references and detector description.
2019-03-19  Ian Brock  <Ian.Brock@cern.ch>
	* Tag as version 07-02-00.
	* February update of publications.
	* Run 2 detector description added.
	* Update ATLAS-SUSY.bib and ATLAS-useful.bib.
2019-02-28  Ian Brock  <Ian.Brock@cern.ch>
	* Tag as version 07-01-01.
	* Update ATLAS-SUSY.bib with correct Errata format and other fixes.
2019-02-22  Ian Brock  <Ian.Brock@cern.ch>
	* Tag as version 07-01-00.
	* January update of publications.
	* Add option backref to atlasbiblatex.
	* Add commented out authorlist to paper template.
	* Add INT note templates for SUSY group. These are a WIP and subject to further change.
2019-01-11  Ian Brock  <Ian.Brock@cern.ch>
	* Tag as version 07-00-00.
	* November update of publications.
	* Rewrite macros for creating cover and title pages.
	Copyright notice should now always be correctly positioned.
	Less space used for title etc.
	Tweaking of offsets should no longer be necessary.
	Warnings about overfull hboxes fixed.

scripts/tex_dollars.py
#! /usr/bin/env python
# Script to replace $...$ and $$...$$ with \(...\) and \[...\]
# Note that this only works if $...$ is on one line.

# Copyright (C) 2002-2020 CERN for the benefit of the ATLAS collaboration

# Changes:
# 2018-10-30 Ian Brock (ian.brock@cern.ch): First version of script.

from __future__ import print_function

import sys, os
import glob
import re
import argparse

def main(args):
    DEBUG = args.debug

    nosubstitutions = ['tikzpicture' , 'verbatim', 'nosubblock']

    # Process all tex and bib filles or just the one specified
    if args.all:
        files = glob.glob('*.tex') + glob.glob('*.bib')
    else:
        files = [args.file]

    # Loop over files to process
    inNosubs = False
    for file in files:
        try:
            fin = open(file)
        except:
            sys.exit('Could not open ' + file)
        fout = open(file + '-new', 'w')
        print('Substitutung $ in', file)

        for line in fin:
            text = line.rstrip()
            if DEBUG > 1: print(text)

            # check for BEGIN of an environment that doesn't want substitutions
            if not inNosubs:
                for nosubstitute in nosubstitutions:
                    if re.match(r'\s*\\begin{\s*' + nosubstitute + '\s*}', text):
                        if DEBUG > 0: print('Nosub started:', text)
                        inNosubs = True

            # check for %\begin{nosubblock}
            if not inNosubs:
                for nosubstitute in nosubstitutions:
                    if re.match(r'\s*%\s*\\begin{\s*' + nosubstitute  + '\s*}', text):
                        if DEBUG > 0: print('Commented nosub started:', text)
                        inNosubs = True

            # check for END of an environment that doesn't want substitutions
            if inNosubs:
                for nosubstitute in nosubstitutions:
                    if re.match(r'\s*\\end{\s*' + nosubstitute + '\s*}', text):
                        if DEBUG > 0: print('Nosub ended:', text)
                        inNosubs = False

            # check for %\end{nosubblock}
            if inNosubs:
                for nosubstitute in nosubstitutions:
                    if re.match(r'\s*%\s*\\end{\s*' + nosubstitute + '\s*}', text):
                        if DEBUG > 0: print('Commented nosub ended:', text)
                        inNosubs = False

            # substitute $$.*$$ with \[.*\]
            if not inNosubs:
                p = re.compile(r'\$\$(.+?)\$\$')
                m = p.search(text)
                if DEBUG > 1: print(p, p.search(text))
                if m:
                    if DEBUG > 0: print('$$...$$ found:', text)
                    text = re.sub(p, '\[\g<1>\]', text)
                    if DEBUG > 1: print(text)

            # substitute $.*$ with \(.*\) 
            # note: this should not match $$.*$$
            if not inNosubs:
                p = re.compile(r'\$(.+?)\$')
                m = p.search(text)
                if DEBUG > 1: print(p, p.search(text))
                if m:
                    if DEBUG > 0: print('$...$ found:', text)
                    text = re.sub(p, '\(\g<1>\)', text)
                    if DEBUG > 1: print(text)

            # Count how many $ are left in the line
            m = re.search(r'\$', text)
            if not inNosubs and m:
                print('There is a $ left:', text)
            if not args.dummy: fout.write(text + '\n')
        fin.close()
        fout.close()
        # Overwrite original file
        if not args.dummy and args.overwrite:
            os.rename(file, file + '.bak')
            os.rename(file + '-new', file)

    return 0

if __name__ == "__main__":
    parser = argparse.ArgumentParser()
    parser.add_argument('-n', '--dummy', help='dummy run', action='store_true')
    parser.add_argument('-a', '--all', help='process alll tex and bib files', action='store_true')
    parser.add_argument('-d', '--debug', help='debug level', type=int, default=0)
    parser.add_argument('-f', '--file', help='filename to process')
    parser.add_argument('-o', '--overwrite', help='overwrite old file (old one renamed to file.bak)', action='store_true')
    args = parser.parse_args()

    returnCode = main(args)


scripts/atlaslatex_2020.sh
#! /bin/bash
# Script to adjust atlaslatex documents to use atlaslatexpath style file.

# Copyright (C) 2002-2020 CERN for the benefit of the ATLAS collaboration.

# Changes:
# 2020-11-23 Ian Brock (ian.brock@cern.ch): First version to get round TeX Live 2020 problem with directories in \usepackage etc.

function cf_files {
    # echo "Compare ${afile} with ${lfile}"
    cmp --silent "$1" "$2"
    cmpStatus=$?
    # echo "cmp status is $cmpStatus" 
    if [ $cmpStatus -eq 0 ]; then
        echo "No change to file $1"
    else
        echo "Running diff on $1 vs $2"
        diff "$1" "$2"
        echo "Will try to copy $2 to $1"
        cp -i "$2" "$1"
    fi
}

DIR=`basename ${PWD}`
echo $DIR

# Check we are in the right directory
if [ -e ${DIR}.tex ]; then
    echo "We are in directory ${PWD}"
else
    echo "We do not appear to be in the main directory: ${PWD}"
    echo "There should be a tex file with the same name as the directory"
    exit 1
fi

# Save a copy of the main file
for lfile in ${DIR}.tex; do
    npath=$(grep -c ATLASLATEXPATH ${lfile})
    echo "ATLASLATEXPATH occurs ${npath} times in ${lfile}"
    if [ $npath -le 0 ]; then
        echo "\\ATLASSLATEXPATH not used - no changes needed"
        exit 0
    fi
    cp ${DIR}.tex ${DIR}.tex.bak
    bfile=${DIR}.tex.bak 
    ATLASLATEXPATHDEF=$(grep -E '^\\newcommand\*\{\\ATLASLATEXPATH\}' ${bfile})
    echo "ATLASLATEXPATH definition: ${ATLASLATEXPATHDEF}"
    if [ -n ${ATLASLATEXPATHDEF} ]; then
        sed -E 's|^\\newcommand\*\{\\ATLASLATEXPATH\}.*|\\RequirePackage\{latex/atlaslatexpath\}|' ${bfile} >${lfile}.tmp1
    else
        echo "ATLASLATEXPATH definition not found in ${lfile}"
        exit 1
    fi        
    sed -E 's|\\ATLASLATEXPATH ||' ${lfile}.tmp1 >${lfile}.tmp2
done

# Old and new main file
for mfile in ${DIR}.tex; do
    ofile=${DIR}.tex
    tfile=${DIR}.tex.tmp2
    cf_files "${ofile}" "${tfile}"
done

# Clean up temporary files
test -e ${lfile}.tmp1 && rm ${lfile}.tmp1
test -e ${lfile}.tmp2 && rm ${lfile}.tmp2


scripts/atlaslatex_update.sh
#! /bin/bash
# Script to update atlaslatex version from the Git master.

# Copyright (C) 2002-2021 CERN for the benefit of the ATLAS collaboration.

# Changes:
# 2018-08-14 Ian Brock (ian.brock@cern.ch): BASENAME should be set correctly if Makefile is overwritten.
# 2019-04-16 Ian Brock (ian.brock@cern.ch): Only overwrite "BASENAME = ..." and not occurences without a space (in help)
# 2020-11-21 Ian Brock (ian.brock@cern.ch): Check for use of \ATLASLATEXPATH and say atlaslatex_2020.sh should be run

# Decide how to clone atlaslatex - ssh is default
ATLASLATEXGIT=ssh://git@gitlab.cern.ch:7999/atlas-physics-office/atlaslatex.git
BRANCH=''
while getopts shkd opt; do
    case $opt in
        s)
            ATLASLATEXGIT=ssh://git@gitlab.cern.ch:7999/atlas-physics-office/atlaslatex.git
            ;;
        h)
            ATLASLATEXGIT=https://gitlab.cern.ch/atlas-physics-office/atlaslatex.git
            ;;
        k)
            ATLASLATEXGIT=https://:@gitlab.cern.ch:8443/atlas-physics-office/atlaslatex.git
            ;;
        d)
            BRANCH=devel
            ;;
        \?)
            echo "$0 [-h] [-k] [-s]"
            echo "  -h use https GitLab link"
            echo "  -k use krb5 GitLab link"
            echo "  -s use ssh GitLab link"
            echo "  -d use devel branch instead of master"
            exit 1
            ;;
    esac
done

# Check we are in the right directory
DIR=$(basename ${PWD})
if [ -e ${DIR}.tex ]; then
    echo "We are in directory ${PWD}"
else
    echo "We do not appear to be in the main directory: ${PWD}"
    echo "There should be a tex file with the same name as the directory"
    exit 1
fi

# Remove temporary directory if it exists
test -d tmp-atlaslatex && rm -r tmp-atlaslatex

# Clone the ATLAS LaTeX Git repository.
# Switch to devel branch for testing
if [ -n "${BRANCH}" ]; then
    git clone ${ATLASLATEXGIT} tmp-atlaslatex
    cd tmp-atlaslatex 
    git checkout devel
    cd ..
else
    git clone --depth 1 ${ATLASLATEXGIT} tmp-atlaslatex
fi

function cf_files {
    # echo "Compare ${afile} with ${lfile}"
    cmp --silent "$1" "$2"
    cmpStatus=$?
    # echo "cmp status is $cmpStatus" 
    if [ $cmpStatus -eq 0 ]; then
        echo "No change to file $1"
    else
        echo "Running diff on $1 vs $2"
        diff "$1" "$2"
        echo "Will try to copy $2 to $1"
        cp -i "$2" "$1"
        # Make sure file is executable
        if [ ${1: -3} == ".sh" ]; then
            echo "Making $1 executable"
            chmod u+x $1
        fi
    fi
}

# Self-update scripts first
test -d scripts || mkdir scripts
scriptupdate=0
for lfile in scripts/atlaslatex_update.sh scripts/atlaslatex_2020.sh; do
    afile=tmp-atlaslatex/scripts/$(basename $lfile)
    if [ -e ${lfile} ]; then
        cmp --silent ${lfile} ${afile}; cmpStatus=$?
        echo "Comparing ${lfile} with ${afile}"
        # echo "Status is ${cmpStatus}"
        if [ $cmpStatus -eq 0 ]; then
            echo "No change to file ${lfile}"
        else
            scriptupdate=1
            cf_files "${lfile}" "${afile}"
            echo "+++ ${lfile} updated. You should now run ${lfile}"
        fi
    else
        scriptupdate=1
        cp ${afile} ${lfile}
        # Make sure file is exectuable
        chmod u+x ${lfile}
        echo "+++ ${lfile} updated. You should now run ${lfile}"
    fi
done
if [ $scriptupdate -eq 1 ]; then
    # Remove temporary directory
    rm -rf tmp-atlaslatex
    exit 1
fi

# Class and style files
for lfile in latex/*.cls latex/*.sty; do
    afile=tmp-atlaslatex/latex/$(basename $lfile)
    cf_files "${lfile}" "${afile}"
done

# atlaslatexpath.sty should be there
if [ -e latex/atlaslatexpath.sty ]; then
    lfile=latex/atlaslatexpath.sty
    afile=tmp-atlaslatex/latex/atlaslatexpath.sty
    cf_files "${lfile}" "${afile}"
else
    echo "Copying atlaslatexpath.sty to latex directory"
    cp tmp-atlaslatex/latex/atlaslatexpath.sty latex/
fi

# Bibliography files
for lfile in bib/*.bib; do
    afile=tmp-atlaslatex/bib/$(basename $lfile)
    cf_files "${lfile}" "${afile}"
done

# Logos
for lfile in logos/*; do
    afile=tmp-atlaslatex/logos/$(basename $lfile)
    cf_files "${lfile}" "${afile}"
done

# Makefile
for lfile in Makefile; do
    BASENAME=$(grep '^BASENAME.*=' ${lfile})
    echo "BASENAME is $BASENAME"
    afile=tmp-atlaslatex/$(basename $lfile)
    cf_files "${lfile}" "${afile}"
    # Assume definition of BASENAME is of the form BASENAME =
    sed -i '.bak' -e "s/BASENAME[ \t]+=.*/${BASENAME}/" ${lfile}
    # The folllowing should only change the first occurence of BASENAME =..., but has not been tested everywhere
    # sed -i '.bak' -e "0,/${BASENAME}/ s/BASENAME[ \t]+=.*/${BASENAME}/" ${lfile}
done

# Acknowledgements - if the directory exists
if [ -d acknowldegements ]; then
    for lfile in acknowledgements/*.tex acknowledgements/*.bib; do
        afile=tmp-atlaslatex/acknowledgements/$(basename $lfile)
        cf_files "${lfile}" "${afile}"
    done
fi

# Remove temporary directory
rm -rf tmp-atlaslatex

# 2020-11-21 ATLASLATEXPATH should no longer be used
mfile=${DIR}.tex
if [ $(grep -c ATLASLATEXPATH ${DIR}.tex) -gt 0 ]; then
    echo "*** IMPORTANT ***"
    echo "*** If you have not already done so, please run scripts/atlaslatex_2020.sh"
    echo "*** This adapts your main tex file to avoid problmes with TeX Live 2020"
    echo "*** It replaces the use of \ATLASLATEXPATH with a style file latex/atlaslatexpath.sty"
    echo "*** IMPORTANT ***"
fi
